src/backend/distributed/shared_library_init.c

```diff
@@ -217,6 +218,13 @@ static const struct config_enum_entry explain_analyze_sort_method_options[] = {
 	{ NULL, 0, false }
 };
 
+
+static const struct config_enum_entry executor_cost_model[] = {
+	{ "none", EXECUTOR_COST_MODEL_NONE, false },
+	{ "current-execution-stats", EXECUTOR_COST_MODEL_CURRENT_EXECUTION_STATS, false },
```

Comment:
vague thought: I wonder whether it would make sense to distinguish between connection pool models like "slow start" vs "cost-based" vs "single connection" vs "force parallelism"

Reply:
I think that sounds like a good one, and let's rename to that.

Though, I'll leave adding other strategies to separate PRs. The ones that I can think of:

```
dynamic: this pr
slow-start: slow star
single: sequential execution
per-shard: force max query parallelization
force-executor-pool-size: similar to force max_query parallelization but upto max_executor_pool_size
```

In the long term, we can drop all other GUCs